cranelift-sizeless-vector.md

```diff
@@ -0,0 +1,46 @@
+# Summary
+
+This RFC proposes a way to handle flexible vector types as specified at: https://github.com/WebAssembly/flexible-vectors
+
+[summary]: #summary
+
+The proposal is to introduce new sizeless vector types into Cranelift, that:
+- Express a vector, with a lane type and size, but a target-defined number of lanes.
+- Are denoted by prefixing the lane type with 'sv' (sizeless vector): svi16, svi32, svf32, etc...
+- Have a minimum width of 128-bits, meaning we can simply map to existing simd-128 implementations.
+
+# Motivation
+[motivation]: #motivation
+
+Flexible vectors are likely coming to WebAssembly so we should support the current spec. This is current path forward to support vectors that are wider than 128-bits.
+Rust is also currently starting to use LLVM's ScalableVectorType and, as a target backend, Cranelift could support those directly with a sizeless vector type.
+
+# Proposal
+[proposal]: #proposal
+
+We can add sizeless vector types by modifying existing structs to hold an extra bit of information to represent the sizeless nature:
+
+- The new types don't report themselves as vectors, so ty.is\_vector() = false, but are explicitly reported via ty.is\_sizeless\_vector().
+- The TypeSet and ValueTypeSet structs gain a bool to represent whether the type is sizeless.
+- TypeSetBuilder also gains a bool to control the building of those types.
+- At the encoding level, a bit is used to represent whether the type is sizeless, this bit has been taken from the max number of vector lanes supported, so they'd be reduced to 128 from 256.
```

Comment:
Please take it from the special types range instead.

Reply:
Updated in latest version.